### PR TITLE
feat: implement a Movie Page 

### PR DESCRIPTION
**Pull request by sofyahreksoftware**
I've implemented a dynamic Movie Page that updates in real-time when a user clicks on a movie tile. Upon clicking, the page retrieves detailed information about the movie from the API using the movie’s ID. This includes comprehensive details like the release date, production location, movie description, as well as information about the cast and crew.

### CODE REVIEW

**Review by wroblewskisoftware — APPROVED** [review]
Hey Sofia!

Great job! I can see that you put a lot of effort into this code, and I really like it. I have a few suggestions that might make it even better. I hope you find them helpful:

1. Styling Naming Conventions: I noticed that you use the `$` sign in your naming conventions, e.g., `$main`, `$people`. It seems like this sign is unnecessary.

2. Modifier Invocation: In some places, you use the notation `$main={$main}`. I think `main` alone would suffice.

3. Colors in Box-Shadow: I see you are using direct color values in the box-shadow property. It might be a good idea to move these colors to your theme.js file for consistency. Similarly, consider defining additional screen widths in media queries in your theme.js file for consistency.

4. Font-Weight Notation: In some instances, you use numerical values for font-weight, and in others, you use words. It would be good to decide on one format for consistency.

5. Headings: In a few places, you use the `<h3>` tag instead of `<h2>`.

6. Optional Chaining in MoviePage: In the MoviePage component, you use the optional chaining operator `?.` quite extensively. Is it necessary in all these instances?

7. Styled-Component PosterContainer: In background-image, you used gradients. Perhaps using an inner shadow `box-shadow: inset` would simplify this part of the styling.

8. fetchDataStatus Variable in movieSlice.js: I also have a suggestion regarding the movieSlice.js file: I noticed the absence of a `fetchDataStatus` variable. Introducing such a variable would be very helpful to clearly indicate the state of data fetching from the API. This variable could provide insights into whether data is currently being fetched, if it has been successfully retrieved, or if an error occurred during the fetching process.

I hope these suggestions are helpful. If you have any questions, feel free to ask!

Once again, congratulations on the great work!

Best regards,
Radek

**Comment by sofyahreksoftware** [comment]
Hi @wroblewski82 😊, 

1. In styled-components, the dollar sign $ before an attribute name is used as a convention to indicate that the attribute should not be passed to the DOM as an HTML property. Instead, its value is used for dynamic styling within the component. You can read more about it [here](https://styled-components.com/docs/api#transient-props). So, I think we should keep props with dollar sign before so that our code complies with the latest conventions.
2. According to the above, the `**$main**` attribute is applied to the entire tile only when it's used on a movie page, not on a list. The condition `**$main={$main}**` is used to check whether `**$main**` is true. If only the `**$main**` attribute is passed, it will always return true, making it impossible to differentiate the styling between a movie tile on a movie page and one on a movie list.
3. Ok, thanks. It's a good idea. I'll move colors and media breakpoints to `**theme.js**`!
4. I will correct it!
5. I think that we should change movie tile header `h2` everywhere`:) And title on movie page on poster as `**h1**` What do you think about it?
6. I don't understand what you mean. Could you please clarify which code part exactly you are talking about?
7. I think if i change gradient to box-shadow it won't look exactly like in Figma, that may spoil the quality.
8. Thanks! I will implement this variable🤗

File: src/features/MoviePage/index.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                  personName={person.name}
+                />
+              ))}
+            </StyledList>
+          </Article>
+
+          <Article>
+            <StyledHeader as="h2">Crew</StyledHeader>
+            <StyledList $people>
+              {credits?.crew?.map((person) => (
```
Comment (wroblewskisoftware) [inline src/features/MoviePage/index.js]: Hi Sofia,

In reference to the sixth point of my code review, I wanted to highlight the section of the code where the use of multiple null-coalescing operators (?.) is prevalent.

Comment (sofyahreksoftware) [reply src/features/MoviePage/index.js]: Thanks! I have already removed `?` operator for all unjustified cases😊

**Pull request merged**